### PR TITLE
Fix historic_agg_v2 TypeError issue #205

### PR DESCRIPTION
**Pull request by Devination**
Please let me know if there are any changes you would like me to make.

### CODE REVIEW

File: alpaca_trade_api/polygon/rest.py [hunk 1]
```diff
@@ -134,21 +135,19 @@ def historic_agg_v2(self, symbol, multiplier, timespan, _from, to,
         path_template = '/aggs/ticker/{symbol}/range/{multiplier}/' \
                         '{timespan}/{_from}/{to}'
         if isinstance(_from, int):
-            path = path_template.format(symbol=symbol,
+            pass
```
Comment (shlomikushchi) [inline alpaca_trade_api/polygon/rest.py:138]: so, if we don't do anything if it's an int we don't need the if

File: alpaca_trade_api/polygon/rest.py [hunk 2]
```diff
@@ -134,21 +135,19 @@ def historic_agg_v2(self, symbol, multiplier, timespan, _from, to,
...
+        elif isinstance(_from, datetime.datetime):
```
Comment (shlomikushchi) [inline alpaca_trade_api/polygon/rest.py:139]: if we do take care of that case, we should also check for `datetime.date`

File: alpaca_trade_api/polygon/rest.py [hunk 3]
```diff
@@ -134,21 +135,19 @@ def historic_agg_v2(self, symbol, multiplier, timespan, _from, to,
...
+            _from = _from.date().isoformat()
+            to = to.date().isoformat()
```
Comment (shlomikushchi) [inline alpaca_trade_api/polygon/rest.py:141]: this entire code segment should be in a method called: format_date()
which will make sure _from and to are in the right format.

**Comment by Devination** [comment]
Your suggestions are good, especially the format_date() method which I'm sure would be useful in other parts of the API I haven't looked at yet.

As far as selecting only 1 date format for the API, that's a design decision for Alpaca. Feel free to discard my pull request if that's what Alpaca wants. Personally I'd prefer to add the flexibility in the API to make it as easy as possible to work with and avoid repeated conversion code in all users' projects.

It's possible that allowing multiple formats might cause confusion but I think people would understand fairly quickly that the API is flexible once they see 2 different examples of different date formats being used.

**Comment by umitanuki** [comment]
I would say it's probably common to accept multiple data types especially this comes with pandas background. I found it useful to have this flexibility myself. We need comprehensive documentation to make it clear what is accepted and how they are converted, both in docstring and README. Can you make that change here before merging?

File: alpaca_trade_api/polygon/rest.py (hunk 3 above)
Comment (umitanuki) [reply alpaca_trade_api/polygon/rest.py:141]: Also, I would handle each parameter separately instead of assuming both are in the same type.

**Comment by shlomikushchi** [comment]
was merged to https://github.com/alpacahq/alpaca-trade-api-python/tree/historic_agg_v2_date_type to make the fixes required locally and then merge to master with: https://github.com/alpacahq/alpaca-trade-api-python/pull/210

**Pull request merged**